tools/m1_utils/update_dynamic.sh

```diff
@@ -0,0 +1,21 @@
+set -e
+
+# Note: this program is the simplest version of patching a dynamic library
+# Other code should handle making frameworks, etc
+patch() {
+     EXTDIR=$(mktemp -d)
+     OUTD="$PWD"
+     pushd $EXTDIR
+     lipo "$FWF" -thin arm64 -output "$OF.ar" || cp "$FWF" "$OF.ar"
+
+     # FIXME: Versions should be input from the build system
+     xcrun vtool -arch arm64 -set-build-version 7 11.0 11.0 -replace -output "$OUTD/$OF" "$OF.ar"
```

Comment:
I remember having to force codesign the output binary here due to restrictions of Apple silicon:
```
xcrun codesign --force --sign - "$OUTD/$OF"
```

...but this could cause the builds to not be reproducible per @keith's [article](https://www.smileykeith.com/2021/10/05/codesign-m1/). Did you run into this as well?

Reply:
> I remember having to force codesign the output binary here due to restrictions of Apple silicon:

This is handled here by the Bazel bundling rules, [rules_apple](https://github.com/bazelbuild/rules_apple/blob/master/apple/internal/partials/framework_import.bzl#L181) - mainly the that code interfaces with the output    artifacts of the `import_middleman` applies an ad-hoc code signature. Pretty nice!

A reproducibility situation can problematic one for Bazel depending on how the outputs from actions are consumed. In some cases, the bundling rules we use here Bazel already assume this because developers [might have a different identity on each machine](https://github.com/bazelbuild/rules_apple/blob/master/apple/internal/partials/framework_import.bzl#L181). Because of many factors, I don't have issues with the reproducibility problems  in that article yet. It'd be ideal to get rules_apple codesign reproducible - if there's any action items we should file a bazelbuild issue.